### ISSUE TITLE
Annoying `BrokenPipeError` traceback being printed when using dev server

### ISSUE THREAD

**Issue by dragoncoder047** [getpelican/pelican#2988]
- [X] I have read the [Filing Issues](https://docs.getpelican.com/en/latest/contribute.html#filing-issues) and subsequent “How to Get Help” sections of the documentation.
- [X] I have searched the [issues](https://github.com/getpelican/pelican/issues?q=is%3Aissue) (including closed ones) and believe that this is not a duplicate.

- **OS version and name**: Raspberry Pi OS (latest version)
- **Python version**: 3.10.2
- **Pelican version**: 4.7.2
- **theme**: my own concotion
- **plugins**: (all from [plugins org](https://github.com/pelican-plugins)) *but I don't think this is a plugin problem*
  - `pelican-share-post`
  - `pelican-sitemap`
  - `pelican-related-posts`
  - `minchin.pelican.plugins.nojekyll`
  - `pelican-read-more`
  - `pelican-jinja2content`
  - `pelican-series`
  - `pelican-more-categories`
- **Link to your site**: not public yet :( sorry

## Issue

I created my site using `pelican-quickstart`. When I use `make devserver`, it builds fine (no errors) and starts the server. I go to the local IP it serves on and my site loads perfectly fine - no 404s or 500s in the browser console. But in the terminal that Pelican is running in, I get a lot of this junk:

```
---------------------
Exception occurred during processing of request from ('192.168.1.158', 52146)
Traceback (most recent call last):
File "/usr/local/lib/python3.10/socketserver.py", line 316, in _handle_request_noblock
    self.process_request(request, client_address)
File "/usr/local/lib/python3.10/socketserver.py", line 347, in process_request
    self.finish_request(request, client_address)
File "/usr/local/lib/python3.10/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
File "/usr/local/lib/python3.10/http/server.py", line 653, in __init__
    super().__init__(*args, **kwargs)
File "/usr/local/lib/python3.10/socketserver.py", line 747, in __init__
    self.handle()
File "/usr/local/lib/python3.10/http/server.py", line 427, in handle
    self.handle_one_request()
File "/usr/local/lib/python3.10/http/server.py", line 415, in handle_one_request
    method()
File "/usr/local/lib/python3.10/site-packages/pelican/server.py", line 87, in do_GET
    server.SimpleHTTPRequestHandler.do_GET(self)
File "/usr/local/lib/python3.10/http/server.py", line 660, in do_GET
    self.copyfile(f, self.wfile)
File "/usr/local/lib/python3.10/http/server.py", line 860, in copyfile
    shutil.copyfileobj(source, outputfile)
File "/usr/local/lib/python3.10/shutil.py", line 198, in copyfileobj
    fdst_write(buf)
File "/usr/local/lib/python3.10/socketserver.py", line 826, in write
    self._sock.sendall(b)
BrokenPipeError: [Errno 32] Broken pipe
---------------------
```

I really have no idea what's going on here.

In case it matters I am using Firefox 91.6.0esr for raspberry pi.

This doesn't affect site build or load at all but since it is printing a traceback I thought it would be worthwhile to report.

**Comment by avaris** [comment]
I don't know the exact situation on your end but that error usually comes when a client terminates connection before full response is sent. IOW, your browser requested something but killed connection before it got the response. You can probably ignore this but if you want to dig deeper, it is likely your browser (or an extension) that is stopping a request prematurely. Keep in mind, `python` server (and in turn `pelican` server) is rather simple in its operation. So, there is not much to fix here in that regard.

**Comment by dragoncoder047** [comment]
Great, thanks!

I have like 10 CSS files and 4 fonts and sometimes it happens on all 15 requests... What do I put for my `LOG_FILTER` to suppress this?

**Comment by egberts** [comment]
Well, it is actually a RichHandler bug, of which I filed over there.  Your output is missing the first traceback which contains the RichHandler.  But I got the same first RichHandler traceback on two different application as you did so I knew the apriori condition.

What is happening is the UNIX pipe was interrupted (Ctrl-C, prematurely closed, or disk error; in your case, a network socket interruption) while the logging.log_handler delegated the STDOUT to RichHandler package.  

Without the RichHandler, the default `logging` package properly redirected its closed STDOUT to `/dev/null` and ate up the rest of the outputs so that there is normally none of that extraneous tracebacks flooding the console screen.

Keep in mind, your traceback is harmless and your apps continues running normally.

https://github.com/Textualize/rich/issues/3400
